modules/apps/object/object-test/src/testFunctional/macros/ObjectModelBuilder.macro

```diff
@@ -1,5 +1,91 @@
 definition {
 
+	macro addObjectFolderViaUi {
+		Click(locator1 = "ObjectAdmin#ADD_OBJECT_FOLDER");
+
+		Type(
+			locator1 = "ObjectAdmin#ADD_FOLDER_LABEL",
+			value1 = ${folderLabel});
+
+		Type(
+			locator1 = "ObjectAdmin#DETAILS_NAME_CUSTOM_OBJECT",
+			value1 = ${fieldName});
+
+		Click(locator1 = "ObjectAdmin#CREATE_FOLDER_BUTTON");
+	}
+
+	macro deleteFolder {
+		Click(locator1 = "ObjectAdmin#DELETE_FOLDER");
+
+		SelectFrame(locator1 = "ObjectCustomViews#VIEW_MODAL_CONTENT");
+
+		Type(
+			locator1 = "ObjectAdmin#PLACEHOLDER_DELETE_OBJECT",
+			value1 = ${objectName});
+
+		Click(locator1 = "Form#SIDEBAR_DELETE_FIELD_MODAL_DELETE_BUTTON");
+	}
+
+	macro editFolderPermissionsInModelBuilder {
+		Click(locator1 = "ObjectAdmin#EDIT_FOLDER_PERMISSIONS");
+
+		ObjectCustomViews.selectAnyPermissions(
+			analyticsAdministrator = ${analyticsAdministrator},
+			guest = ${guest},
+			key_permissionName = ${permissionsName},
+			owner = ${owner},
+			permissionsName = ${permissionsList},
+			portalContentReviewer = ${portalContentReviewer},
+			powerUser = ${powerUser},
+			publicationsUser = ${publicationsUser},
+			siteContentReviewer = ${siteContentReviewer},
+			siteMember = ${siteMember},
+			user = ${user});
+	}
+
+	macro editObjectFolderViaUi {
+		Click(locator1 = "ObjectAdmin#FOLDER_ACTIONS");
+
+		if (${optionSetValue} == "Edit Value and ERC") {
+			Click(locator1 = "ObjectAdmin#EDIT_LABEL_AND_ERC");
+		}
+
+		if (isSet(newLanguage)) {
+			ObjectAdmin.changeLabelLanguage(newLanguage = ${newLanguage});
+		}
+
+		if (isSet(labelName)) {
+			CreateObject.editLabel(labelName = ${labelName});
+		}
+
+		if (isSet(externalReferenceCode)) {
+			Type(
+				key_externalReferenceCode = ${externalReferenceCode},
+				locator1 = "ObjectAdmin#EXTERNAL_REFERENCE_CODE_FOLDER");
+		}
+
+		if (${optionSetValue} == "Folder Permissions") {
+			ObjectModelBuilder.editFolderPermissionsInModelBuilder(
+				analyticsAdministrator = ${analyticsAdministrator},
+				guest = ${guest},
+				key_permissionName = ${permissionsName},
+				owner = ${owner},
+				permissionsName = ${permissionsList},
+				portalContentReviewer = ${portalContentReviewer},
+				powerUser = ${powerUser},
+				publicationsUser = ${publicationsUser},
+				siteContentReviewer = ${siteContentReviewer},
+				siteMember = ${siteMember},
+				user = ${user});
+		}
+
+		if (${optionSetValue} == "Delete Folder") {
+			ObjectModelBuilder.deleteFolder(objectName = ${objectName});
+		}
```

Comment:
@thaiscabrals should we remove these as these are already covered in the previous macros?